src/hiero_sdk_python/consensus/topic_create_transaction.py

```
@@ -122,122 +38,52 @@ def set_auto_renew_period(self, seconds: Union[Duration, int]) -> "TopicCreateTr
+            Exception: If the transaction submission fails or receives an error response.
         """
-        return _Method(
-            transaction_func=channel.topic.createTopic,
-            query_func=None
-        )
+        response = client.topic_stub.createTopic(transaction_proto)
+
+        if response.nodeTransactionPrecheckCode != ResponseCode.OK:
+            error_code = response.nodeTransactionPrecheckCode
+            error_message = ResponseCode.get_name(error_code)
+            raise Exception(f"Error during transaction submission: {error_code} ({error_message})")
```

Ruff (0.15.1)

[warning] Create your own exception

(TRY002)